File: app/src/main/java/com/example/olgadominguez/todoachieved/task/model/TodoTask.java
package com.example.olgadominguez.todoachieved.task.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

@Entity
public class TodoTask {
    @PrimaryKey
    private Long id;

    @NonNull
    private String text;
    private Long createdDate;
    private Long date;

    @Ignore
    public TodoTask(Long id, @NonNull String text, Long createdDate, Long date) {
        this.id = id;
        this.text = text;
        this.createdDate = createdDate;
        this.date = date;
    }

    public TodoTask() {

    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @NonNull
    public String getText() {
        return text;
    }

    public void setText(@NonNull String text) {
        this.text = text;
    }

    public Long getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Long createdDate) {
        this.createdDate = createdDate;
    }

    public Long getDate() {
        return date;
    }

    public void setDate(Long date) {
        this.date = date;
    }
}

File: app/src/main/java/com/cottondroid/todoachieved/task/model/TodoTask.kt
package com.cottondroid.todoachieved.task.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.firebase.database.Exclude

@Entity
data class TodoTask(
        @PrimaryKey(autoGenerate = true)
        @Exclude
        val id: Long? = null,
        val serverId: String? = null,
        val text: String? = null,
        val createdDate: Long? = null,
        val date: Long? = null,
        val serverCreatedTimestamp: Long? = null,
        val serverUpdatedTimestamp: Long? = null
)